File: api/internal/domain/model/author.go
package model

import "time"

type Author struct {
	Base

	NameRu string `gorm:"column:name_ru" json:"name_ru"`
	NameAr string `gorm:"column:name_ar" json:"name_ar"`

	ShortNameRu string `gorm:"column:short_name_ru" json:"short_name_ru"`
	ShortNameAr string `gorm:"column:short_name_ar" json:"short_name_ar"`

	AboutRu string `gorm:"column:about_ru" json:"about_ru"`
	AboutAr string `gorm:"column:about_ar" json:"about_ar"`

	BirthDate *time.Time `gorm:"column:birth_date;type:date" json:"birth_date"`
	DeathDate *time.Time `gorm:"column:death_date;type:date" json:"death_date,omitempty"`

	PicturePath string `gorm:"column:picture_path" json:"picture_path,omitempty"`
	Link        string `gorm:"column:link;index:unique" json:"link,omitempty"`
}

File: app/src/lib/models/Author.js
class Author {
    id: number;
    name_ru: string;
    name_ar: string;
    short_name_ru: string;
    short_name_ar: string;
    about_ru: string;
    about_ar: string;
    birth_date: Date;
    death_date: Date;
    picture_path: string;
    link: string;
}

class AuthorCard {
    name: string;
    short_name: string;
    about: string;
}

export {Author, AuthorCard};